framework/src/play/test/FunctionalTest.java

```diff
@@ -377,6 +399,21 @@ public InvocationContext getInvocationContext() {
         }
     }
 
+    private static RuntimeException unwrapOriginalException(final ExecutionException e) {
+        // Check if the original exceptions fits the usual patterns. If yes, throw the very
+        // original runtime exception
+        final Throwable executionCause = e.getCause();
+        if (executionCause != null
+              && (executionCause instanceof JavaExecutionException || executionCause instanceof UnexpectedException)) {
+            final Throwable originalCause = executionCause.getCause();
+            if (originalCause != null && originalCause instanceof RuntimeException) {
+                throw (RuntimeException) originalCause;
```

Comment:
This is bug. Change `throw (RuntimeException) originalCause;` to `return (RuntimeException) originalCause;`